src/scratch/ScratchRuntime.as

```diff
@@ -346,7 +348,7 @@ public class ScratchRuntime {
 
 			SCRATCH::allow3d {
 				// Should we go 3D?
-				if(isGraphicEffectBlock(b))
+				if(!app.isIn3D && isGraphicEffectBlock(b))
```

Comment:
This new check isn't actually necessary: go3d() checks internally. It won't hurt, though.